tests/test_typeinfo.py

```diff
@@ -44,34 +48,164 @@ async def test_fetch_async(aconn, name, status):
     assert info.array_oid == psycopg.adapters.types["text"].array_oid
 
 
-@pytest.mark.parametrize("name", ["nosuch", sql.Identifier("nosuch")])
-@pytest.mark.parametrize("status", ["IDLE", "INTRANS"])
-def test_fetch_not_found(conn, name, status):
+_name = pytest.mark.parametrize("name", ["nosuch", sql.Identifier("nosuch")])
+_status = pytest.mark.parametrize("status", ["IDLE", "INTRANS"])
+
+
+def _fetch_not_found(conn, name, status, monkeypatch, has_to_regtype, info_cls):
+    exit_orig = psycopg.Transaction.__exit__
+
+    if has_to_regtype:
+
+        def exit(self, exc_type, exc_val, exc_tb):
+            assert exc_val is None
+            return exit_orig(self, exc_type, exc_val, exc_tb)
+
+        monkeypatch.setattr(psycopg.Transaction, "__exit__", exit)
+    else:
+        monkeypatch.setattr(TypeInfo, "_has_to_regtype_function", lambda _: False)
     status = getattr(TransactionStatus, status)
     if status == TransactionStatus.INTRANS:
         conn.execute("select 1")
 
     assert conn.info.transaction_status == status
-    info = TypeInfo.fetch(conn, name)
+    info = info_cls.fetch(conn, name)
     assert conn.info.transaction_status == status
     assert info is None
 
 
-@pytest.mark.asyncio
-@pytest.mark.parametrize("name", ["nosuch", sql.Identifier("nosuch")])
-@pytest.mark.parametrize("status", ["IDLE", "INTRANS"])
-async def test_fetch_not_found_async(aconn, name, status):
+@_name
+@_status
+@pytest.mark.crdb_skip("to_regtype")
+def test_fetch_not_found_TypeInfo_to_regtype(conn, name, status, monkeypatch):
+    _fetch_not_found(conn, name, status, monkeypatch, True, TypeInfo)
+
+
+@_name
+@_status
+def test_fetch_not_found_TypeInfo_cast(conn, name, status, monkeypatch):
+    _fetch_not_found(conn, name, status, monkeypatch, False, TypeInfo)
+
+
+@_name
+@_status
+@pytest.mark.crdb_skip("range")
+def test_fetch_not_found_RangeInfo(conn, name, status, monkeypatch):
+    _fetch_not_found(conn, name, status, monkeypatch, True, RangeInfo)
+
+
+@_name
+@_status
+@pytest.mark.crdb_skip("range")
+@pytest.mark.pg(">= 14")
+def test_fetch_not_found_MultirangeInfo(conn, name, status, monkeypatch):
+    _fetch_not_found(conn, name, status, monkeypatch, True, MultirangeInfo)
+
+
+@_name
+@_status
+@pytest.mark.crdb_skip("composite")
+def test_fetch_not_found_CompositeInfo(conn, name, status, monkeypatch):
+    _fetch_not_found(conn, name, status, monkeypatch, True, CompositeInfo)
+
+
+@_name
+@_status
+@pytest.mark.crdb_skip("to_regtype")
+def test_fetch_not_found_EnumInfo_to_regtype(conn, name, status, monkeypatch):
+    _fetch_not_found(conn, name, status, monkeypatch, True, EnumInfo)
+
+
+@_name
+@_status
+def test_fetch_not_found_EnumInfo_cast(conn, name, status, monkeypatch):
+    _fetch_not_found(conn, name, status, monkeypatch, False, EnumInfo)
```

Comment:
Would it be possible to add a new
```python
@pytest.mark.parametrize(                                                             
    "info_cls, has_to_regtype",
    [
        pytest.param(TypeInfo, True, marks=pytest.mark.crdb_skip("to_regtype")),
        pytest.param(TypeInfo, False),
        pytest.param(RangeInfo, True, marks=pytest.mark.crdb_skip("range")),
        pytest.param(MultirangeInfo, True, marks=pytest.mark.crdb_skip("range")),
        pytest.param(CompositeInfo, True, marks=pytest.mark.crdb_skip("composite")),
        pytest.param(EnumInfo, True, marks=pytest.mark.crdb_skip("to_regtype")),
        pytest.param(EnumInfo, False),
    ],
)
```
to the previous `def test_fetch_not_found()` instead of those new test functions?

Reply:
I completely forgot about marks in params. That's what I wanted. Will look at the pr, thanks